### PR TITLE
Update Korean tutorial for using LLMs, and refactor the nested conditional statements in hr_argparser.py

### PR DESCRIPTION
**Pull request by YeonwooSung**
# What does this PR do?

- Update the Korean docs for LLM tutorials for using Mistral7B, not Llama-v1
- Refactor several nested if-else statements for readability

## Before submitting
- [v] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [v] Did you read the [contributor guideline](https://github.com/huggingface/transformers/blob/main/CONTRIBUTING.md#start-contributing-pull-requests),
      Pull Request section?
- [ ] Was this discussed/approved via a Github issue or the [forum](https://discuss.huggingface.co/)? Please add a link
      to it if that's the case.
- [v] Did you make sure to update the documentation with your changes? Here are the
      [documentation guidelines](https://github.com/huggingface/transformers/tree/main/docs), and
      [here are tips on formatting docstrings](https://github.com/huggingface/transformers/tree/main/docs#writing-source-documentation).
- [ ] Did you write any new necessary tests?

## Who can review?

Anyone in the community is free to review the PR once the tests have passed. Feel free to tag
members/contributors who may be interested in your PR.

@stevhliu

### CODE REVIEW

**Comment by amyeroberts** [comment]
cc @ArthurZucker

**Comment by stevhliu** [comment]
Ok for me to update and showcase Mistral7B, but I'll let @ArthurZucker or @amyeroberts comment on the code refactor :)

File: src/transformers/hf_argparser.py [hunk 1]
```diff
@@ -175,8 +175,10 @@ def _parse_dataclass_field(parser: ArgumentParser, field: dataclasses.Field):
         # A variable to store kwargs for a boolean field, if needed
         # so that we can init a `no_*` complement argument (see below)
         bool_kwargs = {}
-        if origin_type is Literal or (isinstance(field.type, type) and issubclass(field.type, Enum)):
-            if origin_type is Literal:
+        origin_type_is_literal = origin_type is Literal
+        field_typs_is_bool = field.type is bool
```
Comment (stevhliu) [inline src/transformers/hf_argparser.py:179]: typo

```suggestion
        field_type_is_bool = field.type is bool
```

Comment (YeonwooSung) [reply src/transformers/hf_argparser.py:179]: Ah, such a silly issue. Thanks for letting me know

**Comment by amyeroberts** [comment]
@YeonwooSung Thanks for opening this PR! 

Could you separate out any code changes in hf parser and image utils into a separate PR?

**Comment by YeonwooSung** [comment]
> @YeonwooSung Thanks for opening this PR!
> 
> Could you separate out any code changes in hf parser and image utils into a separate PR?

Sure, I will!

**Comment by YeonwooSung** [comment]
@amyeroberts I've just rebased my main branch, and will open a separate pull request for the code changes in hf parser and image utils.

**Review by ArthurZucker — APPROVED** [review]
LGTM 🤗

**Review by amyeroberts — APPROVED** [review]
Thanks for updating!

**Comment by YeonwooSung** [comment]
> BTW we could also use a Korean model to generate Korean text from [here](https://huggingface.co/spaces/upstage/open-ko-llm-leaderboard)

I'll try this next time. Thanks :)

**Pull request merged**